crates/bevy_utils/src/lib.rs

```diff
@@ -415,3 +415,22 @@ macro_rules! detailed_trace {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_clone_entity_hash_map() {
```

Comment:
I'm pretty partial to having something similar to `static_assertions` (it's already in our dependency tree), which includes a assertion that will fail compilation if a type does not implement a trait. I think it should be fine to use it as a dev dependency over these tests which actually need to run when running our test suites.

This is a strict non-blocker and we can tackle this in a separate PR.

Reply:
Ah i was thinking of re-using `trybuild` (https://docs.rs/trybuild/latest/trybuild/) since I see that it's already used in bevy somewhere else, but static_assertions seems to have more powerful checks